pipelines.go

```diff
@@ -55,6 +56,60 @@ func (p *Pipeline) Lock() *Pipeline {
 	return p
 }
 
+func (p *Pipeline) ValidateRefIds() error {
+
+	var refidsSet = make(map[string]bool)
+	var requisitestagerefSet = make(map[string]bool)
+
+	//TODO: What should I return if stages do not exists? I think is valid to not have
+	//Check if there are stages, if not then do not process anything
+	if p.Stages != nil && len(p.Stages) > 0 {
+		//Iterate all stages
+		for _, stageMap := range p.Stages {
+
+			//ALL stages should have a redId
+			if _, exists := stageMap["refId"]; !exists {
+				//TODO: I think this should fail since refId is mandarory?
+				return errors.New("refId is a mandatory field for stages")
+			}
+
+			refId := stageMap["refId"].(string)
+
+			if _,exists := refidsSet[refId]; exists {
+				//TODO: Change this error message
+				return errors.New("refId should be unique, currently two or more stages share the same refId")
+			} else {
+				refidsSet[refId] = true
+			}
+
+			//Check requisiteStageRefIds existence
+			if _,exists := stageMap["requisiteStageRefIds"]; exists {
+				requisiteStageRefIds := stageMap["requisiteStageRefIds"].([]interface{})
+
+				for _, val := range requisiteStageRefIds {
+					// Save relationship refId-requisiteStageRefIds
+					if refId == val {
+						//TODO: Change error message
+						return errors.New("refId cannot be dependant of itself (requisiteStageRefIds)")
```

Comment:
Maybe something like `"circular dependency detected, stage with name %q cannot refer to itself"`

Reply:
I like that error message, @noahi what do you think?